store/src/java/com/zimbra/cs/service/mail/SaveProfileImage.java

```diff
@@ -118,9 +99,71 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
                     ZimbraLog.mailbox.error("Exception in closing input stream for upload", e);
                 }
             }
+            if (!updateDBSuccess) {
+                // Databse update failed; revert from LDAP
```

Comment:
updateDBSuccess will be false even if ldap update fails, in that case there is no need to revert from ldap.

Reply:
Done.